pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -1518,11 +1518,52 @@ func rktFetch(url string, localConfigDir string, pullPolicy string) (string, err
 		return "", errors.New(errMsg)
 	}
 
+	err = rktImageExport(imageID, imageUUID, locFilename)
+	if err != nil {
+		return imageID, err
+	}
+
 	// XXX:FIXME - we should run "rkt image ls" and verify image fetch
 	// went thru without errors.
 	return imageID, nil
 }
 
+func rktImageExport(imageHash, imageUUID, locFilename string) error {
+	var aciDir string
+	var aciName string
+	cmd := "rkt"
+	args := []string{
+		"--dir=" + types.PersistRktDataDir,
+		"--insecure-options=image",
+		"image",
+		"export",
+		imageHash,
+	}
+	if strings.HasSuffix(locFilename, "/") {
```

Comment:
Why do u need this? When can we get 2 types in locFilename ??

Reply:
I guess the issue is whether or not there is a trailing / in locFilename.
Makes sense to use filepath.Join for this instead of the HasSuffix.